Inspired by: https://github.com/xapi-project/xen-api/pull/6006

Do not raise when a certificate doesn't exist after a XAPI upgrade.
`sdn-controller-ca.pem` could have been lost in a previous upgrade,
failing reading it would prevent XAPI from starting.

diff --git i/ocaml/xapi/certificates.ml w/ocaml/xapi/certificates.ml
index 33775182a..671957630 100644
--- i/ocaml/xapi/certificates.ml
+++ w/ocaml/xapi/certificates.ml
@@ -278,14 +278,21 @@ end = struct
       (fun (self, record) ->
         let read_fingerprints filename =
           let ( let* ) = Result.bind in
-          let* certificate =
-            Xapi_stdext_unix.Unixext.string_of_file filename
-            |> Cstruct.of_string
-            |> X509.Certificate.decode_pem
-          in
-          let sha1 = pp_fingerprint ~hash_type:`SHA1 certificate in
-          let sha256 = pp_fingerprint ~hash_type:`SHA256 certificate in
-          Ok (sha1, sha256)
+          try
+            let* certificate =
+              Xapi_stdext_unix.Unixext.string_of_file filename
+              |> Cstruct.of_string
+              |> X509.Certificate.decode_pem
+            in
+            let sha1 = pp_fingerprint ~hash_type:`SHA1 certificate in
+            let sha256 = pp_fingerprint ~hash_type:`SHA256 certificate in
+            Ok (sha1, sha256)
+          with
+          | Unix.Unix_error (Unix.ENOENT, _, _) ->
+              Error
+                (`Msg (Printf.sprintf "filename %s does not exist" filename))
+          | exn ->
+              Error (`Msg (Printexc.to_string exn))
         in
         let filename =
           Filename.concat
